fetch/origin/no-cors.window.js

```diff
@@ -76,3 +74,99 @@ promise_test(async function () {
   assert_equals(json[0], "null");
   assert_equals(json[1], "null");
 }, "Origin header and POST navigation");
+
+var NavigationDestination = {
+  SameOrigin: 1,
+  CrossOrigin: 2,
+  properties: {
+    1: {name: "Same-Origin", origin: origins.HTTP_ORIGIN},
+    2: {name: "Cross-Origin", origin: origins.HTTP_REMOTE_ORIGIN},
+  }
+}
```

Comment:
I wonder if instead of this setup, you could do a nested forEach whereby the second forEach iterates over `["same-origin", "cross-origin"]`. Then you could also inline referrerPolicyFunctor and referrerPolicyTestString and the whole thing would be a little easier to grasp.